### PR TITLE
SRCH-2945 update README

### PR DESCRIPTION
**Pull request by MothOnMars**
## Summary
This PR updates the README to use a cleaner, more concise series of steps to set up a development machine.
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review:

#### Functionality Checks

- [x] Code is functional.

- [x] Automated checks pass, if applicable. If Code Climate checks do not pass, explain reason for failures:

- [x] If your changes will be tested manually, you have run `bundle update` and committed your changes to Gemfile.lock. - N/A
 
- [x] You have merged the latest changes from the target branch (usually `master` or `main`) into your branch.
 
- [x] If your target branch is NOT `master` or `main`, specify the reason:
 
- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release #.#.#** matching the release number
 
- [x] You have squashed your commits into a single commit (exception: your PR includes commits with formatting-only changes, such as required by Rubocop)

- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket
 
#### Process Checks

- [x] You have specified an "Assignee", and if necessary, additional reviewers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,13 +14,6 @@ Read our [contributing guidelines](https://github.com/GSA/search-gov/blob/master
 
 Use [RVM](https://rvm.io/) to install the version of Ruby specified in [.ruby-version](/.ruby-version). 
 
-### Gems
-
-Use [Bundler](https://bundler.io/) to install the required gems:
-
-    $ gem install bundler
-    $ bundle install
```
Comment (MothOnMars) [inline README.md:22]: Gem installation has been moved after package installation

File: README.md [hunk 2]
```diff
@@ -92,23 +99,10 @@ Anything listed in the `secret_keys` entry of that file will automatically be ma
...
-
-A few tips when working with asset pipeline:
-
-* Ensure that your asset directory is in the asset paths by running the following in the console:
-
-    > Rails.application.assets.paths
-
-* Find out which file is served for a given asset path by running the following in the console:
-
-    > Rails.application.assets['relative_path/to_asset.ext']
```
Comment (MothOnMars) [inline README.md:111]: I removed this section, as it's not a necessary step for either development or test setup.

File: README.md [hunk 3]
```diff
@@ -254,25 +248,5 @@ Example:
...
-1. Run mongrel/thin
-
-1. Run a few representative SERPs with news items, gov boxes, etc
-
-1. Visit http://localhost:3000/newrelic
-
-1. The database calls view was the most useful one for me. How many extra database calls did your feature introduce? Yes, they are fast, but at 10-50 searches per second, it adds up.
-
-You can also turn on profiling and look into that (see https://newrelic.com/docs/general/profiling-ruby-applications).
-
```
Comment (MothOnMars) [inline README.md:276]: I've moved this to Confluence, with the caveat that it is ancient and untested with our current version of the `newrelic_rpm` gem.

File: README.md (hunk 1 above)
Comment (jamesmadhoc) [reply README.md:22]: Good call.

File: README.md (hunk 2 above)
Comment (jamesmadhoc) [reply README.md:111]: I found this confusing (what is "my asset directory?"), anyway..

File: README.md (hunk 3 above)
Comment (jamesmadhoc) [reply README.md:276]: Also a good call.

**Review by jamesmadhoc — APPROVED** [review]
LGTM.

**Pull request merged**